File: app/src/main/java/be/digitalia/fosdem/model/AlarmInfo.java
package be.digitalia.fosdem.model;

import androidx.room.ColumnInfo;
import androidx.room.TypeConverters;
import be.digitalia.fosdem.db.converters.NullableDateTypeConverters;

import java.util.Date;

public class AlarmInfo {

	@ColumnInfo(name = "event_id")
	private long eventId;
	@ColumnInfo(name = "start_time")
	@TypeConverters({NullableDateTypeConverters.class})
	private Date startTime;

	public AlarmInfo(long eventId, Date startTime) {
		this.eventId = eventId;
		this.startTime = startTime;
	}

	public long getEventId() {
		return eventId;
	}

	public Date getStartTime() {
		return startTime;
	}
}

File: app/src/main/java/be/digitalia/fosdem/model/AlarmInfo.kt
package be.digitalia.fosdem.model

import android.os.Parcelable
import androidx.room3.ColumnInfo
import androidx.room3.ColumnTypeConverters
import be.digitalia.fosdem.db.converters.NullableInstantTypeConverters
import be.digitalia.fosdem.utils.InstantParceler
import kotlinx.parcelize.Parcelize
import kotlinx.parcelize.WriteWith
import kotlin.time.Instant

@Parcelize
data class AlarmInfo(
        @ColumnInfo(name = "event_id")
        val eventId: Long,
        @ColumnInfo(name = "start_time")
        @field:ColumnTypeConverters(NullableInstantTypeConverters::class)
        val startTime: @WriteWith<InstantParceler> Instant?
) : Parcelable